src/shared/mkfs-util.c

```diff
@@ -669,6 +669,9 @@ int make_filesystem(
         if (r == 0) {
                 /* Child */
 
+                if (strv_extend(&env, strv_find_prefix(environ, "SOURCE_DATE_EPOCH=")) < 0)
+                        return log_oom();
```

Comment:
please move this further up, i.e. add this to `env` already in the parent. i.e. the child should not ideally not allocate memory anymore, but just apply what we prepared in the parent.